drivers/spi/spi_nrfx_spis.c

```diff
@@ -227,9 +227,9 @@ static void event_handler(const nrfx_spis_evt_t *p_event, void *p_context)
 {
 	struct spi_nrfx_data *dev_data = p_context;
 	struct device *dev = CONTAINER_OF(dev_data, struct device, data);
-
 	if (p_event->evt_type == NRFX_SPIS_XFER_DONE) {
-		spi_context_complete(&dev_data->ctx, dev, p_event->rx_amount);
+		spi_context_complete(&dev_data->ctx, dev,
+				     MAX(p_event->rx_amount, p_event->tx_amount));
```

Comment:
This value is returned by `spi_transceive()`, and as API specifies, it is the number of received frames: https://github.com/zephyrproject-rtos/zephyr/blob/a5e04ccd4ab976b99f45d106e3de873fc8dd882a/include/zephyr/drivers/spi.h#L562-L566
so it needs to stay as `p_event->rx_amount`.

Reply:
Does it need to stay like that forever or when are points that could be updated?

It is not useable as a Slave transmitter like this. Except if you keep a rx buffer with the size of at least the tx buffer reserved (which is probably fine for a many cases but definitely not for all).

Reply:
The current spi_nrfx_spis driver implementation fulfills SPI API requirements. If there's a need to get the number of transmitted frames, then the API should be first modified/extended to provide such possibility and then the implementation can be aligned accordingly.